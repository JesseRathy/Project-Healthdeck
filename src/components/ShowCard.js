import React, { Component } from 'react' ;
import { Picker, Text } from 'react-native';
import { connect } from 'react-redux' ;
import { cardUpdate, cardCreate } from '../actions';
import { CardStyle, CardSectionStyle, Input, Button } from './common'

class ShowCard extends Component {

    onButtonPress () {
        const { name, bio, pro, image, link } = this.props;
    }

  render () {
  console.log(this.props.myCards);
  const { name, bio, pro, image, link } = this.props.myCards;

    return (
      <CardStyle>
        <CardSectionStyle>
          <Text style={styles.titleStyle}>
              {pro} {": "} {name}
          </Text>
        </CardSectionStyle>

        <CardSectionStyle>
          <Text style={styles.titleStyle}>
              {bio}
          </Text>
        </CardSectionStyle>

        <CardSectionStyle>
          <Text style={styles.titleStyle}>
              {link}
          </Text>
        </CardSectionStyle>

        <CardSectionStyle>
        <Text style={styles.titleStyle}>
            Image should be displayed here having trouble importing it
        </Text>
        </CardSectionStyle>

      </CardStyle>
    );
  }
}
const styles = {
  titleStyle: {
    fontSize:18,
    paddingLeft:15
  }
};
const mapStateToProps = (state) => {
  const { name, bio, pro, image, link } = state.myCards;

  return { name, bio, pro, image, link } ;
}
export default ShowCard;
