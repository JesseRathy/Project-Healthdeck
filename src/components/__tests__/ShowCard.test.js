import { shallow,configure,mount } from 'enzyme';
import React from 'react';
import configureStore from 'redux-mock-store';
import {connect} from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';
import sinon from "sinon";
import ShowCard from '../ShowCard';
//Imports to get Enzyme to work if possible -> will work on later
configure({ adapter: new Adapter() });

//const willMount = sinon.spy();
//const didMount = sinon.spy();
//const willUnmount = sinon.spy();


test('renders correctly', () => {
  const myCard = {
    name:'Test',
    bio:'Smith',
    pro:"https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
    image:"https://i.imgur.com/K3KJ3w4h.jpg",
    link: 'google.ca'
  };

  const wrapper = shallow(<ShowCard myCards= {
  { name: "Smith",
    bio:'Smith',
    pro:"https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
    image:"https://i.imgur.com/K3KJ3w4h.jpg",
    link: 'google.ca'}}/>);
  expect(wrapper).toMatchSnapshot();
  //expect(tree).toMatchSnapshot();
});

  /*I've just created a const of all the values and called the card with these values
  This is a lot cleaner then that other gross way of doing things.*/



  //const wrapper = shallow(<ShowCard/>);

  //expect(wrapper.find('card')).toBeDefined();

  //expect(wrapper).toMatchSnapshot();


  /*test('it puts each item in the card properly', () =>{
    const cardVals = {
      title:'Test',
      artist:'Smith',
      thumbnail_image:"https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
      image:"https://i.imgur.com/K3KJ3w4h.jpg",
      url: 'google.ca'
    };
/*You have to get the props of whatever is mounted
then you gotta call that prop even though you just got it
then you gotta get the prop of that prop, even though you called props to get the props of the wrapper
I literally spent 3 hours trying to figure this out.
This doesn't make any logical sense. Whatever.
--Jesse
    const wrapper = mount(<CardDetail card={cardVals}/>);
    const myProps = wrapper.props();
    /*used this to test log values to see what I was getting
    Now am using it to show all of the card when I test
    console.log(myProps.card);
    //console.log(myProps.card.title);
    expect(myProps.card.title).toEqual('Test');
    expect(myProps.card.artist).toEqual('Smith');
    expect(myProps.card.thumbnail_image).toEqual("https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg");
    expect(myProps.card.image).toEqual("https://i.imgur.com/K3KJ3w4h.jpg");
    expect(myProps.card.url).toEqual('google.ca');
    //console.log(wrapper.contains(CardDetail,'title'));
});
*/

/* Test idea modified from
https://medium.com/@indvinoth/react-native-how-to-test-onpress-event-using-jest-1cb19115863e
thanks!
Note: Doesn't work.... I'll have to figure out how to test the button.
*/
/*test('Button Works Correctly', () => {
  const onPressEvent = jest.fn();
  onPressEvent.mockReturnValue('Link on press');
  const wrapper = shallow(<CardDetail card={"Test","Artist",null,null,"google.ca"} onPress={onPressEvent}/>);

  wrapper.find(Button).first().props().onPress();
  expect(onPressEvent.mock.calls.length).toBe(1);
});*/
