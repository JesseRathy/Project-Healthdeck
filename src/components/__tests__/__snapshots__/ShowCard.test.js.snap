// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShowCard
    myCards={
      Object {
        "bio": "Smith",
        "image": "https://i.imgur.com/K3KJ3w4h.jpg",
        "link": "google.ca",
        "name": "Smith",
        "pro": "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <CardSectionStyle>
          <Component
            style={
              Object {
                "fontSize": 18,
                "paddingLeft": 15,
              }
            }
          >
            https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg
             
            : 
             
            Smith
          </Component>
        </CardSectionStyle>,
        <CardSectionStyle>
          <Component
            style={
              Object {
                "fontSize": 18,
                "paddingLeft": 15,
              }
            }
          >
            Smith
          </Component>
        </CardSectionStyle>,
        <CardSectionStyle>
          <Component
            style={
              Object {
                "fontSize": 18,
                "paddingLeft": 15,
              }
            }
          >
            google.ca
          </Component>
        </CardSectionStyle>,
        <CardSectionStyle>
          <Component
            style={
              Object {
                "fontSize": 18,
                "paddingLeft": 15,
              }
            }
          >
            Image should be displayed here having trouble importing it
          </Component>
        </CardSectionStyle>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Component
            style={
              Object {
                "fontSize": 18,
                "paddingLeft": 15,
              }
            }
          >
            https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg
             
            : 
             
            Smith
          </Component>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
              " ",
              ": ",
              " ",
              "Smith",
            ],
            "style": Object {
              "fontSize": 18,
              "paddingLeft": 15,
            },
          },
          "ref": null,
          "rendered": Array [
            "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
            " ",
            ": ",
            " ",
            "Smith",
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Component
            style={
              Object {
                "fontSize": 18,
                "paddingLeft": 15,
              }
            }
          >
            Smith
          </Component>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Smith",
            "style": Object {
              "fontSize": 18,
              "paddingLeft": 15,
            },
          },
          "ref": null,
          "rendered": "Smith",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Component
            style={
              Object {
                "fontSize": 18,
                "paddingLeft": 15,
              }
            }
          >
            google.ca
          </Component>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "google.ca",
            "style": Object {
              "fontSize": 18,
              "paddingLeft": 15,
            },
          },
          "ref": null,
          "rendered": "google.ca",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Component
            style={
              Object {
                "fontSize": 18,
                "paddingLeft": 15,
              }
            }
          >
            Image should be displayed here having trouble importing it
          </Component>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Image should be displayed here having trouble importing it",
            "style": Object {
              "fontSize": 18,
              "paddingLeft": 15,
            },
          },
          "ref": null,
          "rendered": "Image should be displayed here having trouble importing it",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardSectionStyle>
            <Component
              style={
                Object {
                  "fontSize": 18,
                  "paddingLeft": 15,
                }
              }
            >
              https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg
               
              : 
               
              Smith
            </Component>
          </CardSectionStyle>,
          <CardSectionStyle>
            <Component
              style={
                Object {
                  "fontSize": 18,
                  "paddingLeft": 15,
                }
              }
            >
              Smith
            </Component>
          </CardSectionStyle>,
          <CardSectionStyle>
            <Component
              style={
                Object {
                  "fontSize": 18,
                  "paddingLeft": 15,
                }
              }
            >
              google.ca
            </Component>
          </CardSectionStyle>,
          <CardSectionStyle>
            <Component
              style={
                Object {
                  "fontSize": 18,
                  "paddingLeft": 15,
                }
              }
            >
              Image should be displayed here having trouble importing it
            </Component>
          </CardSectionStyle>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Component
              style={
                Object {
                  "fontSize": 18,
                  "paddingLeft": 15,
                }
              }
            >
              https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg
               
              : 
               
              Smith
            </Component>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
                " ",
                ": ",
                " ",
                "Smith",
              ],
              "style": Object {
                "fontSize": 18,
                "paddingLeft": 15,
              },
            },
            "ref": null,
            "rendered": Array [
              "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
              " ",
              ": ",
              " ",
              "Smith",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Component
              style={
                Object {
                  "fontSize": 18,
                  "paddingLeft": 15,
                }
              }
            >
              Smith
            </Component>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Smith",
              "style": Object {
                "fontSize": 18,
                "paddingLeft": 15,
              },
            },
            "ref": null,
            "rendered": "Smith",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Component
              style={
                Object {
                  "fontSize": 18,
                  "paddingLeft": 15,
                }
              }
            >
              google.ca
            </Component>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "google.ca",
              "style": Object {
                "fontSize": 18,
                "paddingLeft": 15,
              },
            },
            "ref": null,
            "rendered": "google.ca",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Component
              style={
                Object {
                  "fontSize": 18,
                  "paddingLeft": 15,
                }
              }
            >
              Image should be displayed here having trouble importing it
            </Component>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Image should be displayed here having trouble importing it",
              "style": Object {
                "fontSize": 18,
                "paddingLeft": 15,
              },
            },
            "ref": null,
            "rendered": "Image should be displayed here having trouble importing it",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
